tests/test_compute_funs.py

```diff
@@ -424,3 +516,121 @@ def test_compute_grad_p_volume_avg():
     eq = Equilibrium()  # default pressure profile is 0 pressure
     pres_grad_vol_avg = eq.compute("<|grad(p)|>_vol")["<|grad(p)|>_vol"]
     np.testing.assert_allclose(pres_grad_vol_avg, 0)
+
+
+def test_compute_dmerc(DSHAPE_examples, HELIOTRON):
+    eq = Equilibrium()
+    DMerc = eq.compute("Mercier DMerc")["Mercier DMerc"]
+    np.testing.assert_allclose(DMerc, 0, err_msg="should be 0 in vacuum")
+
+    def test(
+        stellarator, name, rho_range=default_range, rtol=default_rtol, atol=default_atol
+    ):
+        eq = EquilibriaFamily.load(load_from=str(stellarator["desc_h5_path"]))[-1]
+        rho, vmec = get_vmec_data(name, "DMerc")
+        grid = get_grid(eq, rho)
+        DMerc = compress(grid, eq.compute("Mercier DMerc", grid=grid)["Mercier DMerc"])
+        all_close(DMerc, vmec, rho, rho_range, rtol, atol)
+
+    test(DSHAPE_examples, "DSHAPE", (0.175, 0.8))
+    test(DSHAPE_examples, "DSHAPE", (0.8, 1), atol=5e-2)
+    test(HELIOTRON, "HELIOTRON", (0.1, 0.275), rtol=11e-2)
```

Comment:
a lot of the test tolerances seem kind of arbitrary, are these just the lowest tolerances that pass? And are we confident its just a resolution or vmec issue?

Reply:
They're not the lowest tolerance that pass, but they are pretty tight. Going measurably lower would require splitting up ranges with different tolerances. That gets annoying to code.
- They are tight enough to catch changes since the Mercier stuff is highly dependent on everything else. In fact, there was a bug in the grid at one point that affected the equilibrium solve. The flux surfaces still matched exactly, yet these tests picked up a difference in the Mercier results.
- I'm confident my implementations are correct. I can't claim it's a vmec issue, but maybe it's normal to not have exact numerical matches because of the different ways things are computed in the code. If anyone finds a bug I'll buy them a smoothie. We mentioned on Monday, the magnitude of B being wrong sometimes. Maybe that could be necessitating the higher tolerances?
- Plot comparisons are in the zip file in the pull request description. W7X matched well too.